pygeos/tests/test_io.py

```diff
@@ -394,42 +401,21 @@ def test_to_wkb_srid():
         (
             pygeos.geometrycollections([pygeos.multipoints([empty_point])]),
             3,
-            NESTED_COLLECTIONZ_NAN_WKB,
+            NESTED_COLLECTION_NAN_WKB,
         ),
-    ],
-)
-def test_to_wkb_point_empty_pre_geos38(geom, dims, expected):
-    actual = pygeos.to_wkb(geom, output_dimension=dims, byte_order=1)
-    # Use numpy.isnan; there are many byte representations for NaN
-    assert actual[: -dims * 8] == expected[: -dims * 8]
-    assert np.isnan(struct.unpack("<{}d".format(dims), actual[-dims * 8 :])).all()
-
-
-@pytest.mark.skipif(
-    pygeos.geos_version < (3, 8, 0), reason="Post GEOS 3.8.0 has 2D empty points"
-)
-@pytest.mark.parametrize(
-    "geom,dims,expected",
-    [
-        (empty_point, 2, POINT_NAN_WKB),
-        (empty_point, 3, POINT_NAN_WKB),
-        (pygeos.multipoints([empty_point]), 2, MULTIPOINT_NAN_WKB),
-        (pygeos.multipoints([empty_point]), 3, MULTIPOINT_NAN_WKB),
-        (pygeos.geometrycollections([empty_point]), 2, GEOMETRYCOLLECTION_NAN_WKB),
-        (pygeos.geometrycollections([empty_point]), 3, GEOMETRYCOLLECTION_NAN_WKB),
         (
-            pygeos.geometrycollections([pygeos.multipoints([empty_point])]),
+            pygeos.geometrycollections([pygeos.multipoints([empty_point_z])]),
             2,
             NESTED_COLLECTION_NAN_WKB,
         ),
         (
-            pygeos.geometrycollections([pygeos.multipoints([empty_point])]),
+            pygeos.geometrycollections([pygeos.multipoints([empty_point_z])]),
             3,
-            NESTED_COLLECTION_NAN_WKB,
+            NESTED_COLLECTIONZ_NAN_WKB,
         ),
     ],
 )
-def test_to_wkb_point_empty_post_geos38(geom, dims, expected):
+def test_to_wkb_point_empty(geom, dims, expected):
     # Post GEOS 3.8: empty point is 2D
     actual = pygeos.to_wkb(geom, output_dimension=dims, byte_order=1)
     # Use numpy.isnan; there are many byte representations for NaN
```

Comment:
[about the line below] I find it a bit strange that the below works, while it is harcoded for 2 NaNs but some expected results have 3 NaNs (it seems the removed pre_geos38 test used `-dims` instead of a harcoded `-2`

(or maybe it would only fail on platforms we are not testing in CI?)